src/Core/Configurations/RuntimeConfigValidator.cs

```diff
@@ -830,11 +830,161 @@ public void ValidateRelationshipsInConfig(RuntimeConfig runtimeConfig, IMetadata
+                        // Validation to ensure that if linking source fields exist that source fields exist as well.
+                        HandleOrRecordException(new DataApiBuilderException(
+                            message: $"Entity: {entityName} has a many to many relationship: {relationshipName} with linking target fields that are not null, " +
+                                $"but target fields that are null.",
+                            statusCode: HttpStatusCode.ServiceUnavailable,
+                            subStatusCode: DataApiBuilderException.SubStatusCodes.ConfigValidationError));
+                    }
+
+                    // When both source and linkingSource fields exist for a many to many relationship their size must match.
+                    if (relationship.SourceFields is not null && relationship.LinkingSourceFields is not null)
+                    {
+                        if (relationship.SourceFields.Length != relationship.LinkingSourceFields.Length)
+                        {
+                            // Validation to ensure that if source and linking source fields exist, that they have the same number of fields. 
+                            HandleOrRecordException(new DataApiBuilderException(
+                                message: $"Entity: {entityName} has a many to many relationship: {relationshipName} with {relationship.SourceFields.Length} " +
+                                    $"source fields defined, but {relationship.LinkingSourceFields.Length} linking source fields defined.",
+                                statusCode: HttpStatusCode.ServiceUnavailable,
+                                subStatusCode: DataApiBuilderException.SubStatusCodes.ConfigValidationError));
+                        }
+                    }
+
+                    // When both target and linkingTarget fields exist for a many to many relationship their size must match.
+                    if (relationship.TargetFields is not null && relationship.LinkingTargetFields is not null)
+                    {
+                        if (relationship.TargetFields.Length != relationship.LinkingTargetFields.Length)
+                        {
+                            // Validation to ensure that if source and linking source fields exist, that they have the same number of fields. 
+                            HandleOrRecordException(new DataApiBuilderException(
+                                message: $"Entity: {entityName} has a many to many relationship: {relationshipName} with {relationship.TargetFields.Length} " +
+                                    $"target fields defined, but {relationship.LinkingTargetFields.Length} linking target fields defined.",
+                                statusCode: HttpStatusCode.ServiceUnavailable,
+                                subStatusCode: DataApiBuilderException.SubStatusCodes.ConfigValidationError));
+                        }
+                    }
+                }
```

Comment:
Given that we have added new validations here, the validations that are happening below this code on line 1022 seem unnecessary.. Can you please check if they are still necessary, remove if not?

Reply:
They are needed, although our validation makes the conditional a bit redundant. To add context, we were checking `if relationship.LinkingSourceFields is null || relationship.SourceFields is null` before verifying foreign key information exists in the DB because we would "fix" a config that only provided one of the linking source fields or the source fields. We now validate that if one exists, so does the other, but we still want to check that the foreign key information is in the DB if one of them (which now guarantees so is the other) is null.